### PR TITLE
[ENG-1367] Add conflict of interest statement for Sloan

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Allow users to add a conflict of interest statement to their preprint info (on the backend at least)

## Changes

- Add `conflict_of_interest_statement` field to preprint model, with auto-gen migration
- Adds necessary logging info to `PreprintLog` model
- Adds to Preprint serializer to allow for updates to the COI statement with tests.
- Adds waffle flag

## QA Notes

There's no user facing element to this yet, so it should be dev tested until the front-end comes out.

## Documentation

I don't know if we have a place for docs of this nature, in any case it's not functional yet, so no user-facing docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1367

### CODE REVIEW

File: api/preprints/serializers.py [hunk 1]
```diff
@@ -287,6 +291,16 @@ def update(self, preprint, validated_data):
             preprint.original_publication_date = validated_data['original_publication_date'] or None
             save_preprint = True
 
+        if 'conflict_of_interest_statement' in validated_data:
+            if flag_is_active(self.context['request'], SLOAN_COI):
```
Comment (brianjgeiger) [inline api/preprints/serializers.py]: I don't believe this is correct. They should be able to edit the COI regardless of the flag being active for the user, they just wouldn't be able to see the results if they don't have the flag active. But this makes me realize that we don't have any waffle switches for the functionality as a whole. More on this below.

File: osf/features.py [hunk 2]
```diff
@@ -8,6 +8,9 @@
 OSF_PREREGISTRATION = 'osf_preregistration'
 OSF_GROUPS = 'osf_groups'
 ENABLE_CHRONOS = 'enable_chronos'
+SLOAN_COI = 'waffle:sloan|coi'
```
Comment (brianjgeiger) [inline osf/features.py]: As mentioned above, these should not be used for the validation section. Instead, we should have an additional **switch** (not flag, and we can add each per PR) for the whole functionality. The front end will use that to know whether to give the edit controls for the users, and we can use it to regulate the above validation.

File: api_tests/preprints/views/test_preprint_detail.py [hunk 3]
```diff
@@ -535,6 +538,34 @@ def test_update_article_doi(self, app, user, preprint, url):
         assert preprint_detail['links']['doi'] == 'https://doi.org/{}'.format(
             new_doi)
 
+    def test_update_conflict_of_interest_statement(self, app, user, preprint, url):
+        update_payload = build_preprint_update_payload(
+            preprint._id,
+            attributes={'conflict_of_interest_statement': 'Owns shares in Closed Science Inc.'}
```
Comment (brianjgeiger) [inline api_tests/preprints/views/test_preprint_detail.py]: Typo; I believe this is the "Closed Science Corporation".

File: api/preprints/serializers.py [hunk 4]
```diff
@@ -104,6 +106,8 @@ class PreprintSerializer(TaxonomizableSerializerMixin, MetricsSerializerMixin, J
     preprint_doi_created = NoneIfWithdrawal(VersionedDateTimeField(read_only=True))
     date_withdrawn = VersionedDateTimeField(read_only=True, allow_null=True)
     withdrawal_justification = HideIfNotWithdrawal(ser.CharField(required=False, read_only=True, allow_blank=True))
+    conflict_of_interest_statement = ser.CharField(required=False, allow_blank=True, allow_null=True)
```
Comment (brianjgeiger) [inline api/preprints/serializers.py]: Should there be a boolean for "is there a coi"?

**Comment by Johnetordoff** [comment]
merged with https://github.com/CenterForOpenScience/osf.io/pull/9280 and approved

**Pull request closed without merging**